### ISSUE TITLE
Performance benefit of task vs. async CE is likely overstated

### ISSUE THREAD

**Issue by cmeeren** [dotnet/docs#34856]
When comparing `async { }  and `task { }`, the docs currently say:

> * The performance of `task {…}` is much better.

I think this is misleading. Sure, if you use `task` and `async` in fairly hot loops, `task` will come out much better. But usually, when using `task` or `async`, you're doing a HTTP call, querying or updating a database, or similar, and only have a handful of `task` or `async` CEs that perform this work. In that case, unless I'm mistaken, the network/IO overhead generally dwarfs any benefit you would get from replacing `async` with `task`.

As the point quoted above currently stands, it may scare newcomers away from `async` without good reason. I therefore suggest the point be rephrased, to more accurately reflect the extent to which (and maybe circumstances under which) performance improves. For example (if this is correct):

> * The overhead of `task {...}` is lower, possibly improving performance if the asynchronous work is very small

---
#### Document Details

⚠ *Do not edit this section. It is required for learn.microsoft.com ➟ GitHub issue linking.*

* ID: e2dea170-a8f4-90f6-e578-e406fb43fef1
* Version Independent ID: 885f9f09-d2ff-112c-6dda-9a31bccc0e8a
* Content: [What's new in F# 6 - F# Guide](https://learn.microsoft.com/en-us/dotnet/fsharp/whats-new/fsharp-6)
* Content Source: [docs/fsharp/whats-new/fsharp-6.md](https://github.com/dotnet/docs/blob/main/docs/fsharp/whats-new/fsharp-6.md)
* Product: **dotnet-fsharp**
* GitHub Login: @cartermp
* Microsoft Alias: **dotnetcontent**

**Comment by BillWagner** [comment]
ping @dsyme for recommendations on a proposed change.

**Comment by dsyme** [comment]
@cmeeren I agree, that's a fair comment.  Would you like to submit an edit to clarify please?
